akka-typed/src/main/scala/akka/typed/ActorSystem.scala

```diff
@@ -75,65 +61,75 @@ abstract class ActorSystem[-T](_name: String) extends ActorRef[T] { this: ScalaA
    * set on all threads created by the ActorSystem, if one was set during
    * creation.
    */
-  def dynamicAccess: DynamicAccess = untyped.dynamicAccess
+  def dynamicAccess: a.DynamicAccess
 
   /**
-   * The ActorRefProvider is the only entity which creates all actor references within this actor system.
+   * A generic scheduler that can initiate the execution of tasks after some delay.
+   * It is recommended to use the ActorContext’s scheduling capabilities for sending
+   * messages to actors instead of registering a Runnable for execution using this facility.
    */
-  def provider: ActorRefProvider = untyped.provider
+  def scheduler: a.Scheduler
 
   /**
-   * The user guardian’s untyped [[akka.actor.ActorRef]].
+   * Main event bus of this actor system, used for example for logging.
    */
-  private[akka] override def untypedRef: akka.actor.ActorRef = untyped.provider.guardian
+  def eventStream: EventStream
 
   /**
-   * Main event bus of this actor system, used for example for logging.
+   * Facilities for lookup up thread-pools from configuration.
    */
-  def eventStream: EventStream = untyped.eventStream
+  def dispatchers: Dispatchers
 
   /**
    * The default thread pool of this ActorSystem, configured with settings in `akka.actor.default-dispatcher`.
    */
-  implicit def executionContext: ExecutionContextExecutor = untyped.dispatcher
+  implicit def executionContext: ExecutionContextExecutor
 
   /**
    * Terminates this actor system. This will stop the guardian actor, which in turn
    * will recursively stop all its child actors, then the system guardian
    * (below which the logging actors reside).
    */
-  def terminate(): Future[Terminated] = untyped.terminate().map(t ⇒ Terminated(ActorRef(t.actor)))
+  def terminate(): Future[Terminated]
 
   /**
    * Returns a Future which will be completed after the ActorSystem has been terminated
    * and termination hooks have been executed.
    */
-  def whenTerminated: Future[Terminated] = untyped.whenTerminated.map(t ⇒ Terminated(ActorRef(t.actor)))
+  def whenTerminated: Future[Terminated]
 
   /**
    * The deadLetter address is a destination that will accept (and discard)
    * every message sent to it.
    */
-  def deadLetters[U]: ActorRef[U] = deadLetterRef
-  lazy private val deadLetterRef = ActorRef[Any](untyped.deadLetters)
+  def deadLetters[U]: ActorRef[U]
+
+  /**
+   * INTERNAL API
+   */
+  def printTree: String
```

Comment:
yeah, make it official, and give it a descriptive name, such as `printActorTree`

something in between internal and public is bad; we have to pay the bin compat price and users don't dare to use it